api/kyverno/v1/resource_spec_types.go

```diff
@@ -19,16 +20,20 @@ type ResourceSpec struct {
 	// Name specifies the resource name.
 	// +optional
 	Name string `json:"name,omitempty"`
+	// Selector allows you to select with labels
+	// +optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
```

Comment:
I realized that `ResourceSpec` is used in both `TargetResourceSpec` (mutate existing) and `GeneratePattern` (generate), can we do something like this?
```
type TargetResourceSpec struct {
	// ResourceSpec contains the target resources to load when mutating existing resources.
	TargetSelector `json:",omitempty"`
        ...
}

type TargetSelector struct{
	
	ResourceSpec `json:",omitempty"`

	// +optional
	Selector *metav1.LabelSelector `json:"selector,omitempty"`
}
```